apps/explorer/src/app/components/txs/details/tx-details-wrapper.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { DATA_SOURCES } from '../../../config';
+import { t, useFetch } from '@vegaprotocol/react-helpers';
+import { TxDetailsOrder } from './tx-order';
+import type { BlockExplorerTransactionResult } from '../../../routes/types/block-explorer-response';
+import type { TendermintBlocksResponse } from '../../../routes/blocks/tendermint-blocks-response';
+import { TxDetailsHeartbeat } from './tx-hearbeat';
+import { TxDetailsLPAmend } from './tx-lp-amend';
+import { TxDetailsGeneric } from './tx-generic';
+import { TxDetailsBatch } from './tx-batch';
+import { TxDetailsChainEvent } from './tx-chain-event';
+
+interface TxDetailsWrapperProps {
+  txData: BlockExplorerTransactionResult | undefined;
+  pubKey: string | undefined;
+  height: string;
+}
+
+export const TxDetailsWrapper = ({
+  txData,
+  pubKey,
+  height,
+}: TxDetailsWrapperProps) => {
+  const {
+    state: { data: blockData },
+  } = useFetch<TendermintBlocksResponse>(
+    `${DATA_SOURCES.tendermintUrl}/block?height=${height}`
+  );
```

Comment:
**nitpick (non-blocking)**: This should have a cache as right now it triggers a refetch every render

Reply:
Moved to #2172

Reply:
first time i see someone nitpick on their own code :)